Encryption.md

```diff
@@ -139,7 +139,12 @@ union EncryptionAlgorithm {
 
 The `AesGcmV1` and `AesGcmCtrV1` structures contain an optional `aad_metadata` field that can 
 be used by a reader to retrieve the AAD string used for file encryption. The maximal allowed
-length of `aad_metadata` is 512 bytes.
+length of `aad_metadata` is 512 bytes. Typically, readers would know the file AAD without relying
+on this field, using instead a method adopted by the organization. For example, the AAD can be the name of 
+the file - if constructed with a timestamp or version number, the AAD will allow readers to make sure they
+work with a correct file version. If organization works with a number of different 
+AAD construction methods, the `aad_metadata` can contain the method ID. In any case, the AAD string itself
+must not be kept in this field - a reader reliance on this AAD will defy the cipher authentication purpose.
```

Comment:
I don't understand this last sentece, could you please explain it?

Reply:
Yep, the wording is not clear enough, I'll re-write this sentence. The point is as follows: say we have a claims processor that needs to run a query on yesterday's data, making sure it had not been replaced with an older data. The query will specify the AAD based on the expected file name: abcd_<yesterday date>. There is no point in keeping this AAD in the file itself - because it  can be tampered with / replaced - but mostly because the reader should know apriori what data version (date) it looks for.

Reply:
I had to do some research to understand this, it turned out I had mistunderstood AAD earlier. I would propose a decription along the following lines: First state how AAD is used _conventionally_ (AAD is not stored alongside the ciphertext, the ciphertext can only be decrypted when the AAD is provided to the decryption algorithm, essentially behaves like a second password but is used for an entirely different purpose). Then explain that in Parquet we do store the AAD as well (emphasizing that it is not the usual practice), because it allows more flexibility:
- Readers that don't care about checking integrity can still read the data without providing an AAD, since they can read it from the file itself.
- Readers that want integrity checking can provide an expected AAD and encryption will fail unless the same AAD was used for encryption (thereby ensuring integrity).
- Readers that provide multiple AADs can first check whether the stored value is amongst the provided AADs and then do the integrity check using that single AAD instead of trying to decrypt the data using all providedf AADs.

Since this does not seem to be the usual practice, please also explain why storing the AAD does not degrade its security.

Reply:
Ah, okay, so reading your description once more, it seems that I misunderstood again. :) The fields should not store the AAD itself, just some way to identify it (select which one to use).